This patch is based on the upstream commit described below, adjusted
for use in the Debian package by Peter Michael Green.

commit 61cc0589a1aa879e63167e4a862c4b7c033eeb57
Author: Chojan Shang <psiace@outlook.com>
Date:   Tue May 4 12:30:04 2021 +0800

    Upgrade to rand 0.8.3
    
    Signed-off-by: Chojan Shang <psiace@outlook.com>

diff --git a/src/map.rs b/src/map.rs
index f9b6ae968..5d1ba60f7 100644
--- a/src/map.rs
+++ b/src/map.rs
@@ -4277,18 +4277,21 @@ mod test_map {
         let mut m = HashMap::new();
 
         let mut rng = {
-            let seed = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
+            let seed = [
+                1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, // and again
+                1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
+            ];
             SmallRng::from_seed(seed)
         };
 
         // Populate the map with some items.
         for _ in 0..50 {
-            let x = rng.gen_range(-10, 10);
+            let x = rng.gen_range(-10..10);
             m.insert(x, ());
         }
 
         for _ in 0..1000 {
-            let x = rng.gen_range(-10, 10);
+            let x = rng.gen_range(-10..10);
             match m.entry(x) {
                 Vacant(_) => {}
                 Occupied(e) => {
@@ -4587,18 +4590,21 @@ mod test_map {
         let mut m = HashMap::new();
 
         let mut rng = {
-            let seed = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
+            let seed = [
+                1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, // and again
+                1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
+            ];
             SmallRng::from_seed(seed)
         };
 
         // Populate the map with some items.
         for _ in 0..50 {
-            let x = rng.gen_range(-10, 10);
+            let x = rng.gen_range(-10..10);
             m.insert(x, ());
         }
 
         for _ in 0..1000 {
-            let x = rng.gen_range(-10, 10);
+            let x = rng.gen_range(-10..10);
             m.entry(x).and_replace_entry_with(|_, _| None);
             check(&m);
         }
diff --git a/tests/set.rs b/tests/set.rs
index 3fc071705..262926c8c 100644
--- a/tests/set.rs
+++ b/tests/set.rs
@@ -8,13 +8,19 @@ fn test_hashset_insert_remove() {
     let mut m: HashSet<Vec<char>> = HashSet::new();
     //let num: u32 = 4096;
     //let tx: Vec<Vec<u8>> = (0..num).map(|i| (i..(16 + i)).collect()).collect();
-    let seed: [u8; 16] = [
+    let seed: [u8; 32] = [
+        130, 220, 246, 217, 111, 124, 221, 189, 190, 234, 121, 93, 67, 95, 100, 43, // again
         130, 220, 246, 217, 111, 124, 221, 189, 190, 234, 121, 93, 67, 95, 100, 43,
     ];
 
     let rng = &mut SmallRng::from_seed(seed);
     let tx: Vec<Vec<char>> = (0..4096)
-        .map(|_| (rng.sample_iter(&Alphanumeric).take(32).collect()))
+        .map(|_| {
+            rng.sample_iter(&Alphanumeric)
+                .take(32)
+                .map(char::from)
+                .collect()
+        })
         .collect();
 
     for _ in 0..32 {
Index: hashbrown/Cargo.toml
===================================================================
--- hashbrown.orig/Cargo.toml
+++ hashbrown/Cargo.toml
@@ -65,7 +65,7 @@ version = "1.0.7"
 version = "1.4"
 
 [dev-dependencies.rand]
-version = "0.7.3"
+version = "0.8.3"
 features = ["small_rng"]
 
 [dev-dependencies.rayon]
